data/jakarta/utils.sh

```diff
@@ -5,125 +5,67 @@ if [ "$(id -u)" != "0" ]; then
     exit 1
 fi
 
-snap_install()
-{
-    local the_snap=$1
-    local the_channel=$2
-    local confinement=$3
+# get the directory of this script
+# snippet from https://stackoverflow.com/a/246128/10102404
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 
-    if [ "$the_snap" = "edgexfoundry" ]; then
-        if [ -n "$confinement" ]; then
-            snap install "$the_snap" --channel="$the_channel" "$confinement"
-        else
-            snap install "$the_snap" --channel="$the_channel"
-        fi
-    else
-        if [ -n "$confinement" ]; then
-            snap install "$the_snap" "$confinement"
-        else
-            snap install "$the_snap"
-        fi
-    fi
-}
+# load the generic utils
+# shellcheck source=/dev/null
+source "$(dirname "$SCRIPT_DIR")/utils.sh"
 
-snap_refresh()
-{
-    local the_snap=$1
-    local the_channel=$2
-    local confinement=$3
-
-    if [ "$the_snap" = "edgexfoundry" ]; then
-        if [ -n "$confinement" ]; then
-            snap refresh "$the_snap" --channel="$the_channel" "$confinement"
-        else
-            snap refresh "$the_snap" --channel="$the_channel"
-        fi
-    else
-        # for refreshing a file snap we need to use install
-        # but snapd still treats it like a refresh
-        if [ -n "$confinement" ]; then
-            snap install "$the_snap" "$confinement"
-        else
-            snap install "$the_snap"
-        fi
-    fi
-}
-
-snap_check_jakarta_svcs()
+snap_check_svcs()
 {
     if [ "$1" = "--notfatal" ]; then
         FATAL=0
     else
         FATAL=1
     fi
```

Comment:
This seems obsolete.

Reply:
I found that it is used in the functions that are moved to the common utils. To make it clear, it is better to pass FATAL directly to the function.

Reply:
Thanks! I find that `--notfatal` for test-refresh-service is redundant. With `--notfatal`, if there are services in the wrong status, this test will still be passed. I suggest removing the `--notfatal` and `FATAL` variable.

Reply:
Removed.